ister_gui.py

```diff
@@ -1223,19 +1223,31 @@ def _reset_network(self, _):
         except FileNotFoundError:
             pass
 
-        try:
-            subprocess.call(['/usr/bin/systemctl', 'restart',
-                             'systemd-networkd', 'systemd-resolved'])
-        except:
-            raise urwid.ExitMainLoop()
+        self._restart_networkd_resolved()
 
-        # give the network a chance to start up again
-        time.sleep(.1)
+        # set necessary flags, then reset static edit text.
         self.static_ready = False
         self.reset = True
         self.set_static_edits()
         raise urwid.ExitMainLoop()
 
+    def _restart_networkd_resolved(self):
+        """Restart the network services then poll systemctl status output until
+        both are back up"""
+        try:
+            subprocess.call(['/usr/bin/systemctl', 'restart',
+                             'systemd-networkd', 'systemd-resolved'])
+            while True:
```

Comment:
step off increment? This is what I was going to do.

```python
for i in range(10):
    out = subprocess.check_output(['/usr/bin/systemctl',
                                    'status',
                                    'systemd-networkd',
                                    'systemd-resolved'])
    if out.decode('utf-8').count('Active: active (running)') == 2:
        # decrement i in case this is successful on the last try
        # with i == 9
        i -= 1
        break

    time.sleep(.1)

    # if == 9 we hit the try limit
    if i == 9:
        raise Exception('Unable to restart network services')

except Exception as err:
    Alert('Error!', err).do_alert()
    raise urwid.ExitMainLoop()
```